components/bpel/org.wso2.carbon.bpel.ui/src/main/java/org/wso2/carbon/bpel/ui/bpel2svg/impl/ElseImpl.java

```diff
@@ -70,4 +83,240 @@ public Element getSVGString(SVGDocument doc) {
         }
         return group1;
     }
+
+    @Override
+    public SVGDimension getDimensions() {
+        if (dimensions == null) {
+            int width = 0;
+            int height = 0;
+            dimensions = new SVGDimension(width, height);
+
+            SVGDimension subActivityDim = null;
+            ActivityInterface activity = null;
+            Iterator<ActivityInterface> itr = getSubActivities().iterator();
+            while (itr.hasNext()) {
+                activity = itr.next();
+                subActivityDim = activity.getDimensions();
+                if (subActivityDim.getWidth() > width) {
+                    width += subActivityDim.getWidth();
+                }
+                height += subActivityDim.getHeight();
+            }
+
+            height += getYSpacing() + getStartIconHeight();
+            width += getXSpacing();
+
+            dimensions.setWidth(width);
+            dimensions.setHeight(height);
+        }
+
+        return dimensions;
+    }
+
+    @Override
+    public void layout(int startXLeft, int startYTop) {
+        if (layoutManager.isVerticalLayout()) {
+            layoutVertical(startXLeft, startYTop);
+        } else {
+            layoutHorizontal(startXLeft, startYTop);
+        }
+    }
+
+    public void layoutVertical(int startXLeft, int startYTop) {
+        int centreOfMyLayout = startXLeft + (dimensions.getWidth() / 2);
+        int xLeft = centreOfMyLayout - (getStartIconWidth() / 2);
+        int yTop = startYTop + (getYSpacing() / 2);
+
+        ActivityInterface activity = null;
+        Iterator<ActivityInterface> itr = getSubActivities().iterator();
+        int childYTop = yTop + getStartIconHeight() + (getYSpacing() / 2);
+        int childXLeft = startXLeft;
+        while (itr.hasNext()) {
+            activity = itr.next();
+            childXLeft = centreOfMyLayout - activity.getDimensions().getWidth() / 2;
+            activity.layout(childXLeft, childYTop);
+            childYTop += activity.getDimensions().getHeight();
+        }
+
+        setStartIconXLeft(xLeft);
+        setStartIconYTop(yTop);
+        setStartIconTextXLeft(startXLeft + BOX_MARGIN);
+        setStartIconTextYTop(startYTop + BOX_MARGIN + BPEL2SVGFactory.TEXT_ADJUST);
+        getDimensions().setXLeft(startXLeft);
+        getDimensions().setYTop(startYTop);
+
+    }
+
+    public void layoutHorizontal(int startXLeft, int startYTop) {
```

Comment:
Fixed and merged 
https://github.com/wso2/carbon-business-process/blob/master/components/bpel/org.wso2.carbon.bpel.ui/src/main/java/org/wso2/carbon/bpel/ui/bpel2svg/impl/ElseImpl.java#L227